source/common/router/header_parser.cc

```diff
@@ -261,7 +261,9 @@ void HeaderParser::evaluateHeaders(Http::HeaderMap& headers,
   }
 
   for (const auto& formatter : headers_to_add_) {
-    const std::string value = formatter.second->format(stream_info);
+    // Remove any invalid characters the formatter may have introduced.
+    const std::string value =
```

Comment:
Instead of removing invalid characters, we should validate the config (especially for header keys) and reject them.

Reply:
I'm particularly worried about something the fuzzer brought up, "%85n" (or generally "%Nn"). absl::FormatTime introduces a newline from this. Taking a look at strftime I think that's the only case we should worry about for the start time formatter, but I put this in after the formatter because I wasn't sure if any of the other formatters (especially ones we don't write) are introducing invalid characters. 

I'll write a regex validator in the config.